### PR TITLE
[Distributed] update Embedding API

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
Update the Embedding API to make it consistent with other components (`DistGraph` and `DistTensor`) and also similar to Pytorch Embedding.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/distributed/sparse_emb.py [hunk 1]
```diff
@@ -13,31 +13,32 @@ class SparseNodeEmbedding:
...
-        g.ndata[name] = DistTensor(g, shape, F.float32, name, part_policy, initializer)
+    def __init__(self, g, shape, name, init_func, part_policy=None):
+        if part_policy is None:
+            part_policy = PartitionPolicy(NODE_PART_POLICY, g.get_partition_book())
 
-        self._tensor = g.ndata[name]
+        self._tensor = DistTensor(g, shape, F.float32, name, init_func, part_policy)
         self._trace = []
 
     def __call__(self, idx):
```
Comment (aksnzhy) [inline python/dgl/distributed/sparse_emb.py:44]: Do we need store gradient on server?

File: python/dgl/distributed/sparse_emb.py [hunk 2]
```diff
@@ -13,31 +13,32 @@ class SparseNodeEmbedding:
     ----------
     g : DistGraph
```
Comment (aksnzhy) [inline python/dgl/distributed/sparse_emb.py:14]: It is weird to pass the graph to Embedding. Actually the embedding does not rely on any graph information.

File: python/dgl/distributed/sparse_emb.py [hunk 3]
```diff
@@ -13,31 +13,32 @@ class SparseNodeEmbedding:
...
-    name : str
-        The name of the embeddings
     shape : tuple of int
         The shape of the embedding. The first dimension should be the number of nodes.
-    initializer : callable
+    name : str
+        The name of the embeddings
+    init_func : callable
         The function to create the initial data.
+    part_policy : PartitionPolicy
```
Comment (aksnzhy) [inline python/dgl/distributed/sparse_emb.py:22]: How to pass this policy to server?

File: python/dgl/distributed/sparse_emb.py (hunk 1 above)
Comment (aksnzhy) [inline python/dgl/distributed/sparse_emb.py:44]: How to delete the old gradient stored in trace?

File: python/dgl/distributed/sparse_emb.py (hunk 2 above)
Comment (zheng-da) [reply python/dgl/distributed/sparse_emb.py:14]: then what should we pass? unless we want to expose kvstore.

File: python/dgl/distributed/sparse_emb.py (hunk 3 above)
Comment (zheng-da) [reply python/dgl/distributed/sparse_emb.py:22]: i think kvstore should manage it.

File: python/dgl/distributed/sparse_emb.py (hunk 1 above)
Comment (zheng-da) [reply python/dgl/distributed/sparse_emb.py:44]: the trace is deleted in optimizer. but there is a bug if the optimizer isn't called, for example, in the inference. this is something we need to fix.

**Comment by aksnzhy** [comment]
@zheng-da Is this PR ready for merge?

**Comment by zheng-da** [comment]
i can be merged. But I like to have some more updates. I think it's better to rename it as DistEmbedding to make its name consistent with other classes (`DistGraph` and `DistTensor`). let's me push the changes.

**Pull request merged**